concrete/src/User/User.php

```diff
@@ -590,46 +590,50 @@ public function refreshUserGroups()
      */
     public function getUserAccessEntityObjects()
     {
-        $app = Application::getFacadeApplication();
-        $session = $app['session'];
-        $validator = $app->make(SessionValidator::class);
-        // Check if the user is loggged in
-        if ($validator->hasActiveSession() && $session->has('uID')) {
-            $req = Request::getInstance();
+        if ($this->accessEntities === null) {
+            $app = Application::getFacadeApplication();
+            $session = $app['session'];
+            $validator = $app->make(SessionValidator::class);
+            // Check if the user is loggged in
+            if ($validator->hasActiveSession() && $session->has('uID')) {
+                $req = Request::getInstance();
+
+                if ($req->hasCustomRequestUser()) {
```

Comment:
I like your changes in this pull request, but I think we should move these lines to the very top. Otherwise "Previous Page as User" functionality is going to break.

This isn't ideal, obviously this is something that should be fully refactored, but until that happened I don't think we can have this code all the way down here.